test/integration/component/test_ldap_autoImport.py

```diff
@@ -0,0 +1,663 @@
+                 """""
+
+    flag = False
+    try:
+        tn = telnetlib.Telnet(ldapConfiguration['hostname'], ldapConfiguration['port'], timeout=15)
+        if tn is not None:
+            tn.set_debuglevel(1)
+            print tn.msg("Connected to the server")
+            cls.debug("Ldap Server is Up and listening on the port %s" % tn.msg("Connected to the server"))
+            flag = True
+            tn.close()
+    except Exception as e:
+        cls.debug(" Not able to reach the LDAP server ,please check the Services on LDAP %s and  exception is %s"
+                  % ((ldapConfiguration['hostname']), e))
+        cls.reason = "Not able to reach the LDAP server ,please check the Services on LDAP %s and exception is %s" \
+                     % ((ldapConfiguration['hostname']), e)
+
+    return flag
+
+@ddt
+class TestLdap1(cloudstackTestCase):
+    """
+    17 test cases in this class.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+
+        #configure ldap
+        """
+
+        :type cls: object
+        """
+        testClient = super(TestLdap1, cls).getClsTestClient()
+        cls.api_client = testClient.getApiClient()
+        cls.services = testClient.getParsedTestDataConfig()
+        cls.cleanup = []
+        cls.delflag = 0
+        cls.reason = ""
+
+        ###
+        cls.apiClient = cls.testClient.getApiClient()
+        cls.acct = createAccount.createAccountCmd()
+        cls.acct.accounttype = 0
+        cls.acct.firstname = cls.services[
+            "configurableData"]["ldap_account"]["firstname"]
+        cls.acct.lastname = cls.services[
+            "configurableData"]["ldap_account"]["lastname"]
+        cls.acct.password = cls.services[
+            "configurableData"]["ldap_account"]["password"]
+        cls.acct.username = cls.services[
+            "configurableData"]["ldap_account"]["username"]
+        cls.acct.email = cls.services[
+            "configurableData"]["ldap_account"]["email"]
+        cls.acct.account = cls.services[
+            "configurableData"]["ldap_account"]["username"]
+        cls.acct.domainid = 1
+
+        if cls.acct.firstname == "" or cls.acct.lastname == "" or cls.acct.password == "" or cls.acct.username == "" \
+                or cls.services["configurableData"]["ldap_configuration"]["ldapUsername"] == "" or \
+                        cls.acct.account == "" or \
+                        cls.services["configurableData"]["ldap_configuration"]["ldapPassword"] == "":
+
+            cls.debug("Please rerun the test by providing values in ldap configuration user details")
+            cls.skipTest("Please rerun the test by providing proper values in configuration file")
+
+        else:
+            cls.delflag = 1
+            cls.acctRes = cls.apiClient.createAccount(cls.acct)
+        #cls.assertEquals(cls.delflag,  1, "LDAP account details are not provided,please check the configuration")
```

Comment:
Please remove code, do not comment it. It can be retrieved if necessary from Git.